packages/app/src/elements/audit.ts

```diff
@@ -0,0 +1,214 @@
+                </pl-button>
+            </section>
+        `;
+    }
+}
+
+@customElement("pl-audit")
+export class Audit extends StateMixin(LitElement) {
+    @state()
+    private _reusedPasswords: ListItem[] = [];
+
+    @state()
+    private _weakPasswords: ListItem[] = [];
+
+    @state()
+    private _compromisedPasswords: ListItem[] = [];
+
+    @queryAll("pl-audit-list-item")
+    private _countElements: HTMLDivElement[];
+
+    static styles = [
+        shared,
+        css`
+            :host {
+                display: block;
+            }
+
+            .counts {
+                display: grid;
+                grid-gap: 1em;
+                margin: 1em;
+                grid-template-columns: repeat(auto-fit, minmax(20em, 1fr));
+            }
+
+            @media (max-width: 700px) {
+                .counts {
+                    grid-template-columns: repeat(auto-fit, minmax(15em, 1fr));
+                }
+            }
+        `,
+    ];
+
+    render() {
+        const { _reusedPasswords, _weakPasswords, _compromisedPasswords } = this;
+
+        // TODO: Remove this
+        console.log("======== rendering!");
+        console.log(
+            JSON.stringify({
+                reusedPasswordsCount: _reusedPasswords.length,
+                weakPasswordsCount: _weakPasswords.length,
+                compromisedPasswordsCount: _compromisedPasswords.length,
+            })
+        );
+
+        return html`
+            <div class="padded layout">
+                <div class="vertical spacing">
+                    <h3>
+                        Here you can see some counts for passwords you should change, as they might be easier to be
+                        easily guessed.
+                    </h3>
+                    <p class="subtle">Click a count to view the affected items.</p>
+                    <div class="counts">
+                        <pl-audit-list-item
+                            .listItems=${_reusedPasswords}
+                            .label=${$l("Reused Passwords")}
+                            .routeParams=${{ auditReused: "true" }}
+                        ></pl-audit-list-item>
+                        <pl-audit-list-item
+                            .listItems=${_weakPasswords}
+                            .label=${$l("Weak Passwords")}
+                            .routeParams=${{ auditWeak: "true" }}
+                        ></pl-audit-list-item>
+                        <pl-audit-list-item
+                            .listItems=${_compromisedPasswords}
+                            .label=${$l("Compromised Passwords")}
+                            .routeParams=${{ auditCompromised: "true" }}
+                        ></pl-audit-list-item>
+                    </div>
+                </div>
+            </div>
+        `;
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+        this.addEventListener("change", () => this.audit());
```

Comment:
Not sure what you're trying to achieve with this? The `@change` event is usually used with user input, of which there is none here.

Reply:
I got this from another component and hadn't gotten around to seeing if it was necessary yet.